src/xdocs/config_coding.xml

```diff
@@ -173,6 +173,87 @@ int[] a2 = new int[]{
       </subsection>
     </section>
 
+    <section name="AvoidDoubleBraceInitialization">
+      <p>Since Checkstyle 8.29</p>
+      <subsection name="Description" id="AvoidDoubleBraceInitialization_Description">
+        <p>
+          Detects double brace initialization.
+        </p>
+        <p>
+          Rationale: Double brace initialization may look cool, but it is considered
+          as anti-pattern and should be avoided.
```

Comment:
This is not just an anti-pattern. This is also the way to a hard-to-detect memory leak. Because the implicitly created anonymous class is *not static*, it holds a reference to the *this pointer of the outer class*. And if a reference to such a collection is stored somewhere outside, its outer class will survive the GC as long as the inner collection.

```java
class outer {
    List<String> list = new ArrayList<String>(){{
        add("foo");
    }};
  // returns more than just a list
  public List<String> getList() {
    return Collections.unmodifiableList(list);
  }
}
```
is actually
```java
class outer {
    List<String> list = new inner(this);

    class inner extends ArrayList<String> {
        outer outerThis;
        {
            add("foo");
        }
        inner(outer outerThis) {
            this.outerThis = outerThis;
        }
    }
}
```

Please add this to the docs. People love explanations.

Reply:
https://www.baeldung.com/java-double-brace-initialization

> Disadvantages of Using Double Braces
> Holds a hidden reference to the enclosing instance, which may cause memory leaks

Reply:
I am familiar with this issue (hidden link)
I thought mentioning it is just a complication of description.

@pbludov @rnveach do you think it makes sense to add code examples, or just description with (or without) link is enough?

Reply:
I feel description with links is good enough. I would look for a more official link then the one I provided, if possible.  It was just the first link that came up in a google search.

Reply:
I like this article: https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/

Reply:
done.

I didn't find any "official" link, so last one (jooq) seems good to me.